Codereview of 1 from 14

Nintenking opened the pull request:
The only class that I put and I ask you to review is "ImageProcessing", in which in collaboration with Thomas Porro from my group we are trying to auto-crop and auto-rotate the image. Every method should have written in their spec the authors so it should be easy for you to find which ones are made by me.
Thank you for your time!
EDIT: I accidentally committed some .ideas files, I have now deleted them

Nintenking committed:
First Commit

Nintenking committed:
Added class "ImageProcessing"

For review

Nintenking committed:
Cleaned the comments

Nintenking committed:
Delete Project.xml

Nintenking committed:
Delete profiles_settings.xml

Nintenking committed:
Delete misc.xml

Nintenking committed:
Delete modules.xml

Nintenking committed:
Delete runConfigurations.xml

Nintenking committed:
Delete vcs.xml

frankplus commented on app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java:
```diff
@@ -0,0 +1,349 @@
+     * @author Thomas Porro(g1), Oscar Garrido(g1)
+     */
+    private void save(Mat matrix, String imageName) {
+        final String directory = Environment.getExternalStorageDirectory() + "/" + Environment.DIRECTORY_PICTURES + "/ImageProcessingTest/" + imageName;
```
Hardcoding could be avoided by moving the path where the image is saved into a final variable of the class

Nintenking replied:
I didn't think about it, that way it should be more understandable compared to this "random" directory. Thank you!

Nintenking replied:
In the end I completely changed the concept of the method: I was saving the file in an internal directory chosen by me, now I'm doing it into the system's default temp-file directory.

frankplus commented on app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java:
```diff
@@ -0,0 +1,349 @@
+     * @throws FileNotFoundException if the imagePath doesn't exist
+     * @author Oscar Garrido (g1)
+     */
+    public Mat conversion(String imagePath) throws FileNotFoundException {
```
There is an incorrect method overloading. There is Bitmap conversion(Mat matrix) and this one.
These two methods do different things one is converting a matrix into a Bitmap and this one is converting a matrix into a Bitmap. This can cause confusion. I suggest to rename them using more explicative names.

Nintenking replied:
I got the same comparison from my other reviewer and you're absolutely right; I'll make sure to change both the methods names so that they differ and they will describe themselves without confusion

frankplus reviewed (commented):
In my opinion the code is sufficiently structured and documented. I haven't analyzed each image processing step because it goes beyond my expertise. A thorough test of the image processing effectiveness in improving the OCR confidence is needed. Good job for now

Nintenking committed:
Update ImageProcessing.java

I copied here the code I wrote for my other reviewer, I should also have fixed the problems you pointed out!

frankplus reviewed (approved):
good job I don't have anything else to say for now
